src/NewsService.Data/NewsRepository.cs

```diff
@@ -40,6 +42,45 @@ public Guid CreateNews(DbNews news)
             return news.Id;
         }
 
+        public List<DbNews> FindNews(FindNewsParams findNewsParams)
+        {
+            if (findNewsParams == null)
+            {
+                throw new Exception("search parameters not passed.");
+            }
+
+            var dbNewsList = _provider.News.AsQueryable();
+
+            if (findNewsParams.AuthorId != null)
+            {
+                dbNewsList = dbNewsList.Where(x => x.AuthorId == findNewsParams.AuthorId);
+            }
+
+            if (findNewsParams.DepartmentId != null)
+            {
+                dbNewsList = dbNewsList.Where(x => x.DepartmentId == findNewsParams.DepartmentId);
+            }
+
+            if (findNewsParams.Pseudonym != null)
+            {
+                dbNewsList = dbNewsList.Where(x => x.Pseudonym == findNewsParams.Pseudonym);
+            }
+
+            if (findNewsParams.Subject != null)
+            {
+                dbNewsList = dbNewsList.Where(x => x.Subject == findNewsParams.Subject);
+            }
+
+            var dbNewsResponse = dbNewsList.ToList();
+
+            if (dbNewsResponse == null)
+            {
+                throw new Exception("News was not found.");
```

Comment:
maybe we have to return an empty list. Why do we throw exc, if all work well

Reply:
no verification is needed here